src/main/java/busTickets/BusTicketValidation.java

```diff
@@ -1,38 +1,49 @@
 package busTickets;
 
+import busTickets.busTicketStorage.BusTicketStorage;
+import busTickets.busTicketStorage.ViolationStatistics;
+
 import java.time.LocalDate;
 
 public class BusTicketValidation {
-
     public static boolean validateBusTicketStartDate(LocalDate busTicketStartDate, BusTicketType busTicketType) {
-        if (busTicketStartDate == null && (busTicketType == BusTicketType.DAY || busTicketType == BusTicketType.WEEK || busTicketType == BusTicketType.MONTH || busTicketType == BusTicketType.YEAR)) {
-            BusTicket.startDateViolation++;
+        if (busTicketStartDate == null && (busTicketType == BusTicketType.DAY || busTicketType == BusTicketType.WEEK || busTicketType == BusTicketType.YEAR)) {
             return false;
         }
 
-        if (busTicketStartDate != null && busTicketStartDate.isAfter(LocalDate.now())) {
-            BusTicket.startDateViolation++;
-            return false;
-        }
 
-        return true;
+        return busTicketStartDate == null || !busTicketStartDate.isAfter(LocalDate.now());
     }
+
     public static boolean validateBusTicketPrice(Integer price){
         if(price == null || price <= 0){
-            BusTicket.priceViolation++;
-            return false;
-        }
-        if(price % 2 != 0){
-            BusTicket.priceViolation++;
             return false;
         }
-        return true;
+        return price % 2 == 0;
     }
+
     public static boolean validateBusTicketType(BusTicketType busTicketType) {
-        if(busTicketType == null){
-            BusTicket.ticketTypeViolation++;
-            return false;
+        return busTicketType == BusTicketType.DAY || busTicketType == BusTicketType.WEEK || busTicketType == BusTicketType.MONTH || busTicketType == BusTicketType.YEAR;
+    }
+
+    //
+    public static void validateBusTicket(BusTicket busTicket, ViolationStatistics violationStatistics) {
```

Comment:
Why here we have violationStatistics as a param?
Looks like we need return it

Reply:
Should i make then fields in ViolationStatistics static since they have to be shared for another classes as well?
And to not pass it as a parameter, instead make methods in ValidationStatistics static and call them

Reply:
No, no need to make fields static, general idea of validateBusTicket method is to get statictics by ticket, so we put as a param BusTicket object and return ViolationStatistics object with all necessary info, all logic of counting violations should be inside this method